301-spinnaker-vmss-or-aks/azuredeploy.json

```diff
@@ -0,0 +1,402 @@
+{
+  "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+  "contentVersion": "1.0.0.0",
+  "parameters": {
+    "devopsDnsPrefix": {
+      "type": "string",
+      "metadata": {
+        "description": "Unique DNS Name for the Public IP used to access the combined Jenkins and Spinnaker Virtual Machine."
+      }
+    },
+    "adminUsername": {
+      "type": "string",
+      "metadata": {
+        "description": "Username for the DevOps Virtual Machine, the Jenkins instance, and the VM Scale Sets deployed by Spinnaker."
+      }
+    },
+    "authenticationType": {
+      "type": "string",
+      "defaultValue": "sshPublicKey",
+      "allowedValues": [
+        "sshPublicKey",
+        "password"
+      ],
+      "metadata": {
+        "description": "Type of authentication to use on the Virtual Machine. SSH key is recommended."
+      }
+    },
+    "sshPublicKey": {
```

Comment:
Let's simplify the use of alternate auth using the pattern from the repo, see: https://github.com/Azure/azure-quickstart-templates/blob/master/100-marketplace-sample/azuredeploy.json#L337

Reply:
@bmoore-msft, yes I noticed this usage and I believe it definitely works for most cases. While please allow me to raise a justification of not applying it to this template: Besides provisioning a VM, we also provision a Jenkins instance that needs the credential of username + custom password. If we follow the suggested pattern, we have to provide an extra parameter placeholder naming something like "jenkins-pwd" if the user uses ssh public key as the credential for the host VM, i.e. in this situation the user has to provide a Jenkins pwd in addition. Following our current style, we believe the parameter logic is easier to understand: You can choose either pwd or ssh public key for your VM, but you always need to provide a pwd for your Jenkins.

Btw, I'd like to mention a thing that days ago there seemed to be a batch update of using PwdOrSshKey covering all templates. After that change, the business of our Spinnaker-on-Azure template broke with the reason of the Jenkins pwd missing.    

@neil-yechenwei, please update the description of the parameter "sshPublicKey". This parameter does not serve Jenkins.

Reply:
@mybayern1974, I've updated the description of sshPublicKey. Thanks.

Reply:
@bmoore-msft, a kindly ping

Reply:
@bmoore-msft , any chance to take a look?